plugin/store/store.go

```diff
@@ -98,12 +98,31 @@ func (ps *Store) SetState(p *v2.Plugin, state bool) {
 }
 
 // Add adds a plugin to memory and plugindb.
-func (ps *Store) Add(p *v2.Plugin) {
+// An error will be returned if there is a collision.
```

Comment:
Also, Pull() checks for duplicate names using `pm.pluginStore.GetByName`. I like how this fails fast, rather than at `Add`, which is pretty late in the function.  `Add` is still valuable to catch the corner cases, where there's a collision in the generated ID.  Lets be consistent on how both collisions are checked in `Pull` and `Create`.